lib/utils/prompt/confirmation.go

```diff
@@ -33,13 +30,15 @@ import (
 // The prompt is written to out and the answer is read from in.
 //
 // question should be a plain sentece without "[yes/no]"-type hints at the end.
-func Confirmation(out io.Writer, in io.Reader, question string) (bool, error) {
+//
+// ctx can be canceled to abort the prompt.
+func Confirmation(ctx context.Context, out io.Writer, in *ContextReader, question string) (bool, error) {
 	fmt.Fprintf(out, "%s [y/N]: ", question)
-	scan := bufio.NewScanner(in)
-	if !scan.Scan() {
-		return false, trace.WrapWithMessage(scan.Err(), "failed reading prompt response")
+	answer, err := in.ReadContext(ctx)
+	if err != nil {
+		return false, trace.WrapWithMessage(err, "failed reading prompt response")
```

Comment:
Nit: This code is quite old but I think all "WrapWithMessage" calls here can be replaced with just regular "Wrap".

Reply:
`Wrap` just calls `WrapWithMessage` if there are any extra arguments.
I don't see any difference between them, other than a bit of overhead